generator/src/main/java/com/scottlogic/deg/generator/walker/rowspec/RowSpecDecisionTreeWalker.java

```diff
@@ -21,9 +24,10 @@ public RowSpecDecisionTreeWalker(RowSpecTreeSolver rowSpecTreeSolver, RowSpecDat
     }
 
     @Override
-    public Stream<DataBag> walk(DecisionTree tree) {
-        return FlatMappingSpliterator.flatMap(
-            rowSpecTreeSolver.createRowSpecs(tree),
-            rowSpecDataBagGenerator::createDataBags);
+    public DataBagStream walk(DecisionTree tree) {
+        return rowSpecTreeSolver.createRowSpecs(tree)
+            .map(rowSpecDataBagGenerator::createDataBags)
+            .reduce((a, b) -> new DataBagStream(Stream.concat(a.stream(), b.stream()), a.isUnique() || b.isUnique()))
+            .orElse(new DataBagStream(Stream.empty(), false));
```

Comment:
Optional thought here - maybe it'd be nice to introduce an enum of `UNIQUE` vs `NOT_UNIQUE`. Without the context of this PR I wouldn't be clear what this boolean value represents.